metadata-app/src/components/MetadataForm/index.js

```diff
@@ -21,92 +22,222 @@ const formatErrors = (errors) =>
   const [alert, setAlert] = useState();
 
+  if (curDatasetId && shouldFetch) {
+    setShouldFetch(false);
+    Api.fetchDataset(curDatasetId, apiUrl, apiKey)
+      .then((result) => {
+        const apiRes = Object.assign({}, result);
+        apiRes.description = result.notes;
+        setFormValues(Object.assign({}, apiRes));
+        setCurDatasetId(apiRes.id);
+      })
+      .catch((e) => {
+        setAlert(
+          <AlertBox
+            type="error"
+            heading="Error loading metadata"
+            message="See the console output for more information on this error."
+          />
+        );
+        console.error('CREATE DATASET ERROR', e); // eslint-disable-line
+        window.scrollTo(0, 0);
+      });
+  }
+
   // render metadata form
   return (
     <div className="grid-container">
       <Navigation currentStep={currentStep} handleSteps={setCurrentStep} />
+      <Heading currentStep={currentStep} />
       {alert}
-      <Formik
-        initialValues={requiredValues}
-        enableReinitialize="true"
-        validateOnChange={false}
-        validateOnBlur={false}
-        onSubmit={(values) => {
-          Api.createDataset(ownerOrg, values, apiUrl, apiKey)
-            .then((res) => {
-              setAlert(<AlertBox type="success" heading="Dataset updated successfully" />);
-              setRequiredValues(res);
-              window.scrollTo(0, 0);
-            })
-            .catch((error) => {
-              const message = JSON.stringify(error);
-              setAlert(<AlertBox type="error" heading="Error saving metadata" message={message} />);
-              console.error('CREATE DATASET ERROR', error); // eslint-disable-line
-              window.scrollTo(0, 0);
-            });
-        }}
-        validationSchema={RequiredMetadataSchema}
-      >
-        {({ values, handleSubmit, errors, isSubmitting, isValidating }) => {
-          return (
-            <div>
-              {currentStep === 0 && (
-                <h1 className="usite-page-title" id="basic-mega-menu">
-                  Required Metadata
-                </h1>
-              )}
-              {currentStep === 1 && (
-                <h1 className="usite-page-title" id="basic-mega-menu">
-                  Additional Metadata
-                </h1>
-              )}
-              {errors && Object.keys(errors).length > 0 && (
-                <div>
-                  <AlertBox
-                    type="error"
-                    heading="This form contains invalid entries"
-                    errors={formatErrors(errors)}
-                  />
-                  <ErrorFocus
-                    errors={errors}
-                    isSubmitting={isSubmitting}
-                    isValidating={isValidating}
-                  />
-                </div>
-              )}
-              <Form onSubmit={handleSubmit}>
-                {currentStep === 0 && (
+
+      {/* ---------- PAGE 2 -- ADDITIONAL METADA ---------- */}
```

Comment:
This seems a bit unwieldy to me, maybe split the individual steps into their own compoenents, then the metadata form could be a switch statement (although I feel like react-router has a better way to handle this, it's been a while so I don't really recall). I think it would also make it easier to test if they were individual components.

```javascript
switch (currentstep) {
  case 0:
    return <MetadataFormStep1 />;
    break;
  case 1:
    return <MetadataFormStep2 />;
    break;
  // ...
}
```

Reply:
@adborden I believe it's kind of separated into own components, see line 146 and 216. Also, it looks like comments are wrong here, eg, line 83 should say "PAGE 1 -- REQUIRED METADATA". I'm going to push a small fix for it.